corehq/util/metrics/tests/utils.py

```diff
@@ -0,0 +1,18 @@
+from collections import defaultdict
+from contextlib import contextmanager
+
+import mock
+
+
+@contextmanager
+def patch_datadog():
+    def record(fn, name, value, tags=None):
+        key = tuple([
+            name,
+            tuple(sorted(tuple(t.split(':')) for t in (tags or []))),
+        ])
```

Comment:
Is this meant to split tag values that contain colons? Or should it behave like this ...

```suggestion
        def get_tag_pairs(tags: list):
            return tuple(sorted(tuple(t.split(':', 1)) for t in tags))
        key = (name, get_tag_pairs(tags or []))
```
... where `get_tag_pairs()` does this?
```python
>>> get_tag_pairs(['tagkey:t:a:g:v:a:l:u:e'])
(('tagkey', 't:a:g:v:a:l:u:e'),)
```